packages/design-picker/src/components/generated-design-picker.tsx

```diff
@@ -77,24 +78,69 @@ const GeneratedDesignPicker: React.FC< GeneratedDesignPickerProps > = ( {
 } ) => {
 	const { __ } = useI18n();
 
+	/* eslint-disable @typescript-eslint/no-unused-vars */
+	const wrapperRef = useRef< HTMLDivElement >( null );
+	const viewmoreRef = useRef< HTMLDivElement >( null );
+	const propertyForThumbnailHeight = '--thumbnail-design-height';
+
+	useEffect( () => {
+		const onResize = () => {
+			const { current: thumnbnailWrapper } = wrapperRef;
+			const { current: viewmoreWrapper } = viewmoreRef;
+			if ( ! thumnbnailWrapper || ! viewmoreWrapper ) {
+				return;
+			}
+
+			const viewportHeight = Math.max(
+				document.documentElement.clientHeight,
+				window.innerHeight || 0
+			);
+			const wrapperMaxHeight =
+				viewportHeight -
+				( thumnbnailWrapper.getClientRects()[ 0 ].y + window.scrollY ) -
+				viewmoreWrapper.getClientRects()[ 0 ].height;
+			const thumbnailHeight = wrapperMaxHeight / 3;
+			const thumbnailActualHeight = Math.min( 300, Math.max( 100, thumbnailHeight ) );
+
+			thumnbnailWrapper.style.setProperty(
+				propertyForThumbnailHeight,
+				`${ thumbnailActualHeight }px`
+			);
+		};
+
+		window.addEventListener( 'resize', onResize );
+		onResize();
+
+		return () => {
+			window.removeEventListener( 'resize', onResize );
+		};
+	}, [] );
+	/* eslint-enable @typescript-eslint/no-unused-vars */
+
 	return (
 		<div className="generated-design-picker">
 			{ heading }
 			<div className="generated_design-picker__content">
 				<div className="generated-design-picker__thumbnails">
-					{ designs &&
-						designs.map( ( design, index ) => (
-							<GeneratedDesignThumbnail
-								key={ design.slug }
-								slug={ design.slug }
-								thumbnailUrl={ getDesignPreviewUrl( design, { language: locale, verticalId } ) }
-								isSelected={ selectedDesign?.slug === design.slug }
-								onPreview={ () => onPreview( design, index ) }
-							/>
-						) ) }
-					<Button className="generated-design-picker__view-more" onClick={ onViewMore }>
-						{ __( 'View more options' ) }
-					</Button>
+					<div className="generated-design-picker__thumbnails-wrapper" ref={ wrapperRef }>
+						{ designs &&
+							designs
+								.slice( 0, 3 )
```

Comment:
Remember that this is okay for testing and we have to remove it as it's handled by D81487-code

Reply:
Noted, I will remove the slicing on the last commit